api/views.py

```diff
@@ -43,8 +43,10 @@ class TeacherLesson(APIView):
   parser_classes = [JSONParser]
 
   def get(self, request, pk):
-    lesson = Lesson.objects.get(pk=pk)
-    return Response(LessonSerializer(lesson).data)
+    teacher = Teacher.objects.get(pk=pk)
+    lessons = teacher.lesson_set.all()
+    
+    return Response(LessonSerializer(lessons, many=True).data)
```

Comment:
so the lesson serializer here takes care of getting the questions and answers as well?

Reply:
Ya take a look at the LessonSerialzier and you can see it also has a questions attribute, which then calls a chain of other serializers to get all the questions and answers.